core/src/Kokkos_Complex.hpp

```diff
@@ -278,11 +278,7 @@ class
     return *this;
   }
 
-  //---------------------------------------------------------------------------
-  // TODO: refactor Kokkos reductions to remove dependency on
-  // volatile member overloads since they are being deprecated in c++20
-  //---------------------------------------------------------------------------
-
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_3
```

Comment:
Marked with 3 for the moment, but of course trivial to change if that's where consensus points